CodeDocumentation/codeDocumentation.md

```diff
@@ -0,0 +1,258 @@
+
+- `unsigned long previousMillis` : Previous time for buzzer.
+
+- `int currentRep` : Number of repetitions for buzzer.
+
+- `const int ledPin` : LED pin.
+
+- Function declarations for buzzer, GPS sensor, motion sensor, etc.
+
+## Setup functions  
+
+```
+void setup(){
+}
+```
+
+This function is a one-time function and will initialize the following elements: 
+
+- Pin and peripheral initialization
+
+- Timer initialization 
+
+- Initialization of Bluetooth, IMU, GPS and SIM modules
+
+- Battery voltage reading. 
+
+## Loop functions  
+
+```
+void loop(){
+}
+```
+
+The void loop() function being the main function of the code, it contains major pieces of code.
+As these pieces of code are numerous, they won't be documented in this document.
+Although, they will be well-commented by the developers, to enhance a clear understanding.
+
+
+## Additional functions  
+
+ The additional functions can be used for a variety of purposes. For example, to retrieve sensor data (such as battery voltage, motion and rotation data), manage time, activate/deactivate GPS and alarm, communicate with a Bluetooth Low Energy (BLE) device, and convert GPS coordinates.
+
+Below is a more detailed description of the additional functions: 
+
+```
+float getBatteryVoltage() {}
+```
+ 
+- This function retrieves the battery voltage. It is useful to know if the battery needs to be charged or not. 
+
+
+```
+float getMotionData() {}
+```
+
+ - This function recovers movement data from the sensor. This makes it possible to measure the intensity of the shock (more or less strong) and to sound the buzzer according to the calculated intensity of the shock.
+
+
+```
+float getRotationData() {}
+```
+
+- Like the function `float getMotionData() `, it retrieves rotation data to make the buzzer sound according to the intensity of rotation. 
+
+```
+void Temps(void) {}
+``` 
+ - This function initializes the seconds and minutes. It can be used to display in the serial monitor when a shock occurs, for example or to initialize timers.
+ 
+```
+void PulseBuzzer(int repetitions, unsigned long durationOn, unsigned long durationOff) {}
+```
+
+- This function allows the buzzer to sound with repetitions and durations that can be defined.
+
+```
+void GPS_ISR() {}
+```
+
+- This function manages the start of GPS position search when required and the transmission of the acquired position when available, depending on the status of GPS activation and the status of the acquired position.
+ 
+```
+void activateGPS() {}
+```
+
+- This function activates the GPS, allowing GPS coordinates to be retrieved when a shock is detected.
+
+```
+void TimerHandler() {}
+```
+- This function launch the ISR timer. 
+
+```
+void SIM_ISR() {}
+```
+
+- Initializes position as TRUE boolean.
+
+```
+void onConnect(BLEDevice central) {}
+```
+
+- This functions allow the bluetooth connection and turn on a led to show if something is connect on bluetooth and if the apparaing bluetooth in on. 
```

Comment:
Use "this function allows the Bluetooth and turns" instead of "this functions allow the bluetooth and turn"
"[...] something is connected on Bluetooth" instead of "[...] something is connect on bluetooth"
Be careful with your Bluetooth uppercases.